File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, string control) 
{
    for(int i=0; i<control.length(); i++)
    {
        switch(control[i])
        {             
            case 'w':
                n++;
                break;
            case 'a':
                n -= 10;
                break;
            case 's':
                n--;
                break;
            case 'd':
                n += 10;
                break; 
        }
    }
    return n;
}



// #include <string>
// #include <vector>
// #include <map>
// using namespace std;

// map <char, int> m = {{'w', 1}, {'s', -1}, {'d', 10}, {'a', -10}};
// int solution(int n, string control) 
// {
//     int answer = n;
//     for(char ch : control)
//         answer += m[ch];
//     return answer;
// }

File: 프로그래머스/0/181926. 수 조작하기 1/수 조작하기 1.cs
using System;

public class Solution {
    public int solution(int n, string control) {
        int answer = 0;
        for(int i=0; i<control.Length; i++) {
            switch(control[i].ToString()) {
                case "w":
                    n += 1;
                    break;
                case "s":
                    n -= 1;
                    break;
                case "d":
                    n += 10;
                    break;
                case "a":
                    n -= 10;
                    break;
            }   
        }
        return n;
    }
}